region/brinstar/pink/Spore Spawn Farming Room.json

```diff
@@ -288,6 +288,38 @@
         }
       }
     },
+    {
+      "link": [2, 1],
+      "name": "Roll Under Door",
+      "entranceCondition": {
+        "comeInRunning": {
+          "speedBooster": true,
+          "minTiles": 26
+        }
+      },
+      "requires": [
+        {"notable": "Roll Under Door"},
+        "canSpeedball"
+      ],
+      "bypassesDoorShell": true,
+      "exitCondition": {
```

Comment:
This will only be able to match with another room that has a comeinwithmockball. Can you enter and immediately stop? If so, there could be another variant that doesnt leave with the speedball

Reply:
Yeah, I'm seeing maybe 3 options here for how to make the strat more broadly applicable as a door lock bypass:

- Add another variant without the exit condition.
- Define a way for `leaveWithMockball` to match with `comeInNormally`.
- Add `comeInWithMockball` strats everywhere, like what we did with `comeInWithSpark` on vertical doors.

Things can get tricky because even though you could stop immediately in the next room, you would still be morphed, and it takes a few frames to unmorph if that's what you need to do. An example would be entering Main Hall; the horizontal speed would not be helpful, and the unmorph would slow you down a bit.

Another example where strats leave in a non-normal way but we currently pretend it is normal: grapple clip door lock bypass strats in Halfie Climb and Colosseum. You spawn in a higher position above the doorway. It could be unsound though as far as I know it hasn't bitten us yet. 

Assuming that a normal exit isn't actually possible with this strat, it's more correct for the `leaveWithMockball` exit condition to be there. We could define a way for `leaveWithMockball` to match with `comeInNormally`, by having it add a few extra heat frames for safety if the next room is heated. You might want that to be possible to override, since it's still not quite the same as a normal entry; e.g. the delay while you're unmorphing could throw off timings on enemy behavior. 

The other alternative, of adding `comeInWithMockball` strats everywhere, would make explicit exactly how it affects things. It's probably the most conceptually clean solution, but it's a lot of relatively meaningless strats, which come with a cost in terms of the work to create them and the clutter they would add.